e2etest/scenario_helpers.go

```diff
@@ -541,11 +541,82 @@ func (scenarioHelper) generateAzureFilesFromList(c asserter, shareURL azfile.Sha
 }
 
 func (s scenarioHelper) enumerateShareFileProperties(a asserter, shareURL azfile.ShareURL) map[string]*objectProperties {
+	var dirQ []azfile.DirectoryURL
+	result := make(map[string]*objectProperties)
+
+	root := shareURL.NewRootDirectoryURL()
+	dirQ = append(dirQ, root)
+	for i := 0; i < len(dirQ); i++ {
+		currentDirURL := dirQ[i]
+		for marker := (azfile.Marker{}); marker.NotDone(); {
+			lResp, err := currentDirURL.ListFilesAndDirectoriesSegment(context.TODO(), marker, azfile.ListFilesAndDirectoriesOptions{})
+			a.AssertNoErr(err)
+
+			// Process the files and folders we listed
+			for _, fileInfo := range lResp.FileItems {
+				fileURL := currentDirURL.NewFileURL(fileInfo.Name)
+				fgpr, err := fileURL.GetProperties(context.TODO())
+				a.AssertNoErr(err)
+
+				// Construct the properties object
+				fileSize := fgpr.ContentLength()
+				creationTime, err := time.Parse(azfile.ISO8601, fgpr.FileCreationTime())
+				a.AssertNoErr(err)
+				lastWriteTime, err := time.Parse(azfile.ISO8601, fgpr.FileLastWriteTime())
+				a.AssertNoErr(err)
+				contentHeader := fgpr.NewHTTPHeaders()
+				h := contentHeaders{
+					cacheControl:       &contentHeader.CacheControl,
+					contentDisposition: &contentHeader.ContentDisposition,
+					contentEncoding:    &contentHeader.ContentEncoding,
+					contentLanguage:    &contentHeader.ContentLanguage,
+					contentType:        &contentHeader.ContentType,
+					contentMD5:         contentHeader.ContentMD5,
+				}
+				fileAttrs := fgpr.FileAttributes()
+				filePermissions := fgpr.FilePermissionKey()
+
+				props := objectProperties{
+					isFolder:           false, // no folders in Blob
+					size:               &fileSize,
+					nameValueMetadata:  fgpr.NewMetadata(),
+					contentHeaders:     &h,
+					creationTime:       &creationTime,
+					lastWriteTime:      &lastWriteTime,
+					smbAttributes:      &fileAttrs, //TODO
```

Comment:
For this, I actually never figured out how attributes should be represented in the test harness code.  There seem to be at least three possibilities:
1. Use an unsigned integer that corresponds to what is natively used (as a set of bitflags) by Windows.
2. Use an array of strings (whole words) which corresponds to what is used by Azure Files.
3. Use our own format in the test harness, which might be a set of single letters (specifically the same set that we use as our command line filter parameters when filtering by attributes) or maybe some other format of our choosing.  I like the conciseness of the command-line format.

Since you're taking over the test harness, this is probably something that you can decide and implement. Basically you'll need to change the type of objectProperties.smbAttributes to match your choice. Then you'll need to ensure the appropriate conversions to and/or from that format when that property is used to interface with Windows and Azure Files. (Windows and Azure Files are the only places that have such attributes and the only places we support the attributes).

If you do _not_ use the same format as on the command line, then you'll need to look at with.smbAttributes in file declarativeWithPropertyProviders.go.  There's a "todo" comment there.